### PR TITLE
refactor: rename `MsgRemoveForeignCoin` `name` field into `zrc20Address` and remove `crosschain` param

### PR DESCRIPTION
**Pull request by ptdatta**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes: #1650  & #1596
    

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [x] Tested CCTX in localnet
- [x] Tested in development environment
- [x] Go unit tests
- [x] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: proto/fungible/tx.proto [hunk 1]
```diff
@@ -69,7 +69,7 @@ message MsgDeployFungibleCoinZRC20Response {
 
 message MsgRemoveForeignCoin {
   string creator = 1;
-  string name = 2;
+  string zrc20_address = 2; //zrc20 address
```
Comment (lumtis) [inline proto/fungible/tx.proto:72]:
```suggestion
  string zrc20_address = 2;
```
unnecessary

File: x/fungible/types/message_remove_foreign_coin.go [hunk 2]
```diff
@@ -11,8 +11,8 @@ var _ sdk.Msg = &MsgRemoveForeignCoin{}
 
 func NewMsgRemoveForeignCoin(creator string, name string) *MsgRemoveForeignCoin {
 	return &MsgRemoveForeignCoin{
-		Creator: creator,
-		Name:    name,
+		Creator:      creator,
+		ZRC20Address: name,
 	}
 }
```
Comment (lumtis) [inline x/fungible/types/message_remove_foreign_coin.go:17]: name should be renamed into zrc20Address

**Review by lumtis — COMMENTED** [review]
Thanks!

We need to run: 
```
make generate
```

And could we add this line in `changlog.md` under `# Unreleased`
```
## Unreleased

- `MsgRemoveForeignCoin`: `Name` field has been renamed in to `ZRC20Address`
```

**Comment by ptdatta** [comment]
> Thanks!
> 
> We need to run:
> 
> ```
> make generate
> ```
> 
> And could we add this line in `changlog.md` under `# Unreleased`
> 
> ```
> ## Unreleased
> 
> - `MsgRemoveForeignCoin`: `Name` field has been renamed in to `ZRC20Address`
> ```

Oops sorry, I previously ran `make proto`

File: x/fungible/types/message_remove_foreign_coin.go (hunk 2 above)
Comment (ptdatta) [reply x/fungible/types/message_remove_foreign_coin.go]: Done

**Comment by lumtis** [comment]
> > Thanks!
> > We need to run:
> > ```
> > make generate
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And could we add this line in `changlog.md` under `# Unreleased`
> > ```
> > ## Unreleased
> > 
> > - `MsgRemoveForeignCoin`: `Name` field has been renamed in to `ZRC20Address`
> > ```
> 
> Oops sorry, I previously ran `make proto`

Does the command complete successfully?
It openapi file get deleted while it shouldn't happen when running the command

**Comment by ptdatta** [comment]
> > > Thanks!
> > > We need to run:
> > > ```
> > > make generate
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And could we add this line in `changlog.md` under `# Unreleased`
> > > ```
> > > ## Unreleased
> > > 
> > > - `MsgRemoveForeignCoin`: `Name` field has been renamed in to `ZRC20Address`
> > > ```
> > 
> > 
> > Oops sorry, I previously ran `make proto`
> 
> Does the command complete successfully? It openapi file get deleted while it shouldn't happen when running the command

No, it just deleted some of the files. So, I again cloned the repo again and made changes to it.

**Comment by lumtis** [comment]
> > > > Thanks!
> > > > We need to run:
> > > > ```
> > > > make generate
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > And could we add this line in `changlog.md` under `# Unreleased`
> > > > ```
> > > > ## Unreleased
> > > > 
> > > > - `MsgRemoveForeignCoin`: `Name` field has been renamed in to `ZRC20Address`
> > > > ```
> > > 
> > > 
> > > Oops sorry, I previously ran `make proto`
> > 
> > 
> > Does the command complete successfully? It openapi file get deleted while it shouldn't happen when running the command
> 
> No, it just deleted some of the files. So, I again cloned the repo again and made changes to it.

Ok, it seems the swagger file is still deleted in the PR

**Comment by ptdatta** [comment]
> > > > > Thanks!
> > > > > We need to run:
> > > > > ```
> > > > > make generate
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > And could we add this line in `changlog.md` under `# Unreleased`
> > > > > ```
> > > > > ## Unreleased
> > > > > 
> > > > > - `MsgRemoveForeignCoin`: `Name` field has been renamed in to `ZRC20Address`
> > > > > ```
> > > > 
> > > > 
> > > > Oops sorry, I previously ran `make proto`
> > > 
> > > 
> > > Does the command complete successfully? It openapi file get deleted while it shouldn't happen when running the command
> > 
> > 
> > No, it just deleted some of the files. So, I again cloned the repo again and made changes to it.
> 
> Ok, it seems the swagger file is still deleted in the PR

Sorry, Added it.

File: changelog.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * `zetaclientd start` : 2 inputs required from stdin
+* `MsgRemoveForeignCoin`: `Name` field has been renamed in to `ZRC20Address`
```
Comment (kingpinXD) [inline changelog.md:6]: Please move this under `Refactor`

Comment (ptdatta) [reply changelog.md]: Done

**Comment by lumtis** [comment]
Looks like there are some build issues @ptdatta

**Review by kingpinXD — APPROVED** [review]

**Comment by lumtis** [comment]
Closing as it seems no longer active. Please let us know if you want to take it back. Thanks for the PR.

**Pull request closed without merging**